### PR TITLE
Containers refresh

### PR DESCRIPTION
**Pull request by mheon**
Add a new command to manually reset container state, allowing us to more gracefully handle potentially-breaking changes in the database. This would run as a post-install script in package managers, and perform a task somewhat similar to a daemon restart (except without the daemon) - containers go down, have their state updated to reflect changes in the current version, and come back up.

Also make some changes to the way we handle CGroups to address bugs I found while writing this.

### CODE REVIEW

**Comment by mheon** [comment]
This is WIP pending tests and manpages, but I wanted to get the code out for review

**Comment by rhatdan** [comment]
I would prefer to make this a subcommand of podman containers

# podman containers refresh.

I am currently working on 

#podman containers cleanup.

File: libpod/container_internal.go [hunk 1]
```diff
@@ -681,6 +697,32 @@ func (c *Container) stop(timeout uint) error {
 	return c.cleanup()
 }
 
+// Internal, non-locking function to pause a container
```
Comment (TomSweeneyRedHat) [inline libpod/container_internal.go]: This is a nice change, but I would have preferred it being in a separate PR.  A) to make the PR review easier, B) so you can search for the pause/unpause change more easily in git many months hence.

Comment (mheon) [reply libpod/container_internal.go]: I can split the changes out trivially, this was just the first thing written that required an internal pause/unpause.

**Comment by mheon** [comment]
Tests are in, removing WIP label

**Comment by baude** [comment]
LGTM ... though I have been out for a while.  If I had a vote, I would prefer something like podman --refresh rather than an actual subcommand that will only be used once in a while.  But I might be too late to the party for that.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request closed without merging**